crates/primitives/chain_config/src/chain_config.rs

```diff
@@ -189,6 +215,21 @@ impl ChainConfig {
         serde_yaml::from_str(&config_str).context("While deserializing chain config")
     }
 
+    /// Verify that the chain config is valid for block production.
+    pub fn precheck_block_production(&self) -> anyhow::Result<()> {
+        // block_time != 0 implies that n_pending_ticks_per_block != 0.
+        if self.sequencer_address == ContractAddress::default() {
+            bail!("Sequencer address cannot be 0x0 for block production.")
+        }
+        if self.block_time.as_millis() == 0 {
+            bail!("Block time cannot be zero for block production.")
+        }
+        if self.pending_block_update_time.as_millis() == 0 {
```

Comment:
we should also check that pending_block_update_time is < block_time right ?

Reply:
good idea yea